packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -696,7 +728,24 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         const builder = new SummaryTreeBuilder();
-        builder.addBlob(`${gcBlobPrefix}_root`, JSON.stringify(gcState));
+        const sortedGCState: IGarbageCollectionState = generateSortedGCState(gcState);
+        const newBlob = JSON.stringify(sortedGCState);
+
+        if (
+            this.latestSerializedGCBlobs.get(gcBlobRootKey) === newBlob &&
+            this.latestSerializedGCBlobs.size === 1 &&
```

Comment:
The size check is important. I was thinking about how to handle scenario where we start writing multiple blobs and this is the answer :-).

Reply:
I was concerned about multiple blobs too. Now that I think about it. it doesn't matter if we have multiple blobs. We will always read from a snapshot, so if a newer client writes multiple blobs, and nothing changes, we will combine all the blobs into one before we serialize it.

Reply:
The interesting case is when the GC state has not changed since the last summary. A new client writes multiple blobs and the combined state is the same as the one generated by an older client that wants to write a single blob.

Reply:
So as long as we are pulling the GC data from the snapshot, now that I think about it, we don't have to worry about this case. So if a future client basically says there are no changes, then the new client will load from the snapshot and have the gc blob serialized in its format. So the summary itself may look different, but the snapshots should be the same.